### PR TITLE
a few details in combinat, as suggested by ruff

### PR DESCRIPTION
**Pull request by fchapoton**
this is fixing a few things in combinat, as suggested by the very fast linter `ruff`

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.

### CODE REVIEW

File: src/sage/combinat/partition.py [hunk 1]
```diff
@@ -4201,8 +4201,8 @@ def zero_one_sequence(self):
             sage: all(Partitions().from_zero_one(mu.zero_one_sequence()) == mu for n in range(10) for mu in Partitions(n))
             True
         """
-        tmp = [self[i]-i for i in range(len(self))]
-        return ([Integer(not (i in tmp)) for i in range(-len(self)+1,self.get_part(0)+1)])
+        tmp = [self[i] - i for i in range(len(self))]
```
Comment (dcoudert) [inline src/sage/combinat/partition.py]: May be `tmp` should be a set and not a list. I'm not sure if it is interesting here, and I agree it's not the object of this PR.

**Review by dcoudert — COMMENTED** [review]
Otherwise, LGTM.

**Review by dcoudert — APPROVED** [review]
LGTM.

**Comment by vbraun** [comment]
merge conflict

**Comment by fchapoton** [comment]
back to positive after trivial resolution of conflict

**Pull request merged**